### PR TITLE
Add adc_read_dt()

### PR DESCRIPTION
**Pull request by nixward**
This was missing from the ADC Devicetree API set.

### CODE REVIEW

File: drivers/sensor/current_amp/current_amp.c [hunk 1]
```diff
@@ -28,7 +28,7 @@ static int fetch(const struct device *dev, enum sensor_channel chan)
 		return -ENOTSUP;
 	}
 
-	ret = adc_read(config->port.dev, &data->sequence);
+	ret = adc_read_dt(config->port.dev, &data->sequence);
```
Comment (kartben) [inline drivers/sensor/current_amp/current_amp.c:31]: Hopefully CI catches this one?

```suggestion
	ret = adc_read_dt(config->port, &data->sequence);
```

Comment (nixward) [reply drivers/sensor/current_amp/current_amp.c]: Fixed 👍

File: include/zephyr/drivers/adc.h [hunk 2]
```diff
@@ -681,6 +681,21 @@ static inline int z_impl_adc_read(const struct device *dev,
...
+ * @brief Set a read request from a struct adc_dt_spec.
+ *
+ * @param spec ADC specification from Devicetree.
+ * @param sequence  Structure specifying requested sequence of samplings.
+ *
+ * @return A value from adc_read().
+ * @see adc_read()
+ */
+static inline int adc_read_dt(const struct adc_dt_spec *spec,
+		       const struct adc_sequence *sequence)
```
Comment (fabiobaltieri) [inline include/zephyr/drivers/adc.h:694]: nit: could you check the indentation? don't think it's a hard rule but it looks like the rest of the file is indented on the parenthesis

**Review by fabiobaltieri — COMMENTED** [review]
Good stuff, just one nit

File: include/zephyr/drivers/adc.h (hunk 2 above)
Comment (nixward) [reply include/zephyr/drivers/adc.h]: Thanks, fixed 👍

**Review by fabiobaltieri — APPROVED** [review]

**Review by aasinclair — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**